training/ilab-wrapper/ilab

```diff
@@ -18,96 +33,115 @@ function verify_range() {
+    # In case of normal user, $XDG_RUNTIME_DIR is used
+    if [[ -f ${XDG_RUNTIME_DIR}/containers/auth.json ]]; then
+        ADDITIONAL_MOUNT_OPTIONS+=("-v" "${XDG_RUNTIME_DIR}/containers/auth.json:/run/containers/0/auth.json")
+    # For root the main /run/containers is used
+    elif [[ -f /run/containers/${UID}/auth.json ]]; then
+        ADDITIONAL_MOUNT_OPTIONS+=("-v" "/run/containers/${UID}/auth.json:/run/containers/0/auth.json")
+    fi
 
-    verify_range "$CURRENT_USER_SUBUID_RANGE" "$CURRENT_USER_NAME"
+    if [[ "$UID" == 0 ]]; then
+        # If we're already running as root, we don't need to map any UIDs
+        IMPERSONATE_CURRENT_USER_PODMAN_FLAGS=()
+    else
+        CURRENT_USER_NAME=$(id --user --name)
+        CURRENT_USER_SUBUID_RANGE=$(awk \
+            --field-separator ':' \
+            --assign current_user="$CURRENT_USER_NAME" \
+            --assign current_uid="$UID" \
+            '$1 == current_user || $1 == current_uid {print $2 ":" $3}' \
+            /etc/subuid)
 
-    IMPERSONATE_CURRENT_USER_PODMAN_FLAGS=("--uidmap" "0:$UID" "--uidmap" "1:$CURRENT_USER_SUBUID_RANGE")
-fi
+        verify_range "$CURRENT_USER_SUBUID_RANGE" "$CURRENT_USER_NAME"
 
-PRESERVE_ENV="VLLM_LOGGING_LEVEL,NCCL_DEBUG,HOME,HF_TOKEN"
-PODMAN_COMMAND=("sudo" "--preserve-env=$PRESERVE_ENV" "podman" "run" "--rm" "-it"
-    "${IMPERSONATE_CURRENT_USER_PODMAN_FLAGS[@]}"
-    "--device" "${CONTAINER_DEVICE}"
-    "--security-opt" "label=disable" "--net" "host"
-    "--shm-size" "10G"
-    "-v" "$HOME:$HOME"
-    "${ADDITIONAL_MOUNT_OPTIONS[@]}"
-    # This is intentionally NOT using "--env" "HOME" because we want the HOME
-    # of the current shell and not the HOME set by sudo
-    "--env" "VLLM_LOGGING_LEVEL"
-    "--env" "HOME"
-    "--env" "NCCL_DEBUG"
-    "--entrypoint" "$ENTRYPOINT"
-    "--env" "HF_TOKEN"
-    "${IMAGE_NAME}")
-
-exec "${PODMAN_COMMAND[@]}" "${PARAMS[@]}"
+        IMPERSONATE_CURRENT_USER_PODMAN_FLAGS=("--uidmap" "0:$UID" "--uidmap" "1:$CURRENT_USER_SUBUID_RANGE")
+    fi
+
+    PRESERVE_ENV="VLLM_LOGGING_LEVEL,NCCL_DEBUG,HOME,HF_TOKEN"
+    PODMAN_ENV_FLAGS=()
+    for ENV_VAR in $(echo "$PRESERVE_ENV" | tr ',' ' '); do
+        # This is intentionally using the "--env VARNAME=$VARNAME" form and not
+        # the "--env VARNAME" form because we want the VARNAME of the current
+        # shell and not the VARNAME set by sudo
+        #
+        # Also we must conditionally set the environment variable, because if
+        # the variable is not set, we don't want to pass an empty value to the
+        # container, this could lead to unexpected behavior from instructlab.
+        #
+        # Also we don't want to simply use sudo --preserve-env=VARNAME,VARNAME,...
+        # because this requires special sudo permissions that are not present
+        # for all users by default.
+        if [[ -n "${!ENV_VAR}" ]]; then
+            PODMAN_ENV_FLAGS+=("--env" "$ENV_VAR=${!ENV_VAR}")
```

Comment:
We can't use the = syntax (see recent fix for explanation https://github.com/containers/ai-lab-recipes/pull/731)

This is no longer needed since the --preserve-env  line above (added in above fix) carries the env from the calling environment into sudo, and then the podman exec will reuse the correct value from there.

Reply:
See comment:

```bash
        # Also we must conditionally set the environment variable, because if
        # the variable is not set, we don't want to pass an empty value to the
        # container, this could lead to unexpected behavior from instructlab.
        #
        # Also we don't want to simply use sudo --preserve-env=VARNAME,VARNAME,...
        # because this requires special sudo permissions that are not present
        # for all users by default.
```